Sources/Player/Player+Metrics.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  Copyright (c) SRG SSR. All rights reserved.
+//
+//  License information is available from the LICENSE file.
+//
+
+import Combine
+import CoreMedia
+import PillarboxCore
+
+private let kMetricsQueue = DispatchQueue(label: "ch.srgssr.player.metrics")
+
+public extension Player {
+    /// Returns a publisher periodically emitting metrics when the player is playing content.
+    ///
+    /// - Parameters:
+    ///   - interval: The interval at which events must be emitted, according to progress of the current time of the timebase.
+    ///   - queue: The queue on which values are published.
+    /// - Returns: The publisher.
+    ///
+    /// Additional non-periodic updates will be published when time jumps or when playback starts or stops.
+    func periodicMetricsPublisher(forInterval interval: CMTime, queue: DispatchQueue = .main) -> AnyPublisher<Metrics?, Never> {
+        queuePlayer.currentItemPublisher()
+            .map { [queuePlayer] item -> AnyPublisher<Metrics?, Never> in
+                guard let item else { return Just(nil).eraseToAnyPublisher() }
+                return Publishers.PeriodicTimePublisher(for: queuePlayer, interval: interval, queue: kMetricsQueue)
+                    .scan(MetricsState.empty) { initial, next in
+                        initial.updated(with: item.accessLog(), at: next) ?? initial
+                    }
+                    .removeDuplicates()
+                    .withPrevious(MetricsState.empty)
+                    .map { $0.current.metrics(from: $0.previous) }
+                    .eraseToAnyPublisher()
+            }
```

Comment:
I wanted to simplify do that: 

```swift
            .compactMap { $0 }
            .map { [queuePlayer] item -> AnyPublisher<Metrics?, Never> in
                Publishers.PeriodicTimePublisher(for: queuePlayer, interval: interval, queue: kMetricsQueue)
                    .scan(MetricsState.empty) { initial, next in
                        initial.updated(with: item.accessLog(), at: next) ?? initial
                    }
                    .removeDuplicates()
                    .withPrevious(MetricsState.empty)
                    .map { $0.current.metrics(from: $0.previous) }
                    .eraseToAnyPublisher()
            }
```

The code above leads to unexpected metrics (increments with negative values).
I tried to write a test, but unfortunately I failed.

Reply:
The `compactMap` is probably not appropriate here. If we have no item we want to have no metrics, so it is quite natural to return `nil` here instead of inhbiting the update because of the `compactMap`.

Reply:
What do you think about adding the following test to avoid any errors in the future?

```swift
func testMetricsWithAnEmptyPlayer() {
    let player = Player()
    expectAtLeastEqualPublished(
        values: [nil],
        from: player.periodicMetricsPublisher(forInterval: .init(value: 1, timescale: 1)),
        timeout: .seconds(1)
    )
}
```

Reply:
Sure. I also introduced similar tests on a development branch for #904.

Reply:
Done. I also added a basic test with a playing player.